plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/Maven2GroupRecipe.groovy

```diff
@@ -0,0 +1,116 @@
+ * trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven.internal.maven2
+
+import javax.annotation.Nonnull
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Provider
+import javax.inject.Singleton
+
+import org.sonatype.nexus.repository.Format
+import org.sonatype.nexus.repository.RecipeSupport
+import org.sonatype.nexus.repository.Repository
+import org.sonatype.nexus.repository.Type
+import org.sonatype.nexus.repository.maven.internal.context.CoordinatesHandler
+import org.sonatype.nexus.repository.maven.internal.group.MavenGroupHandler
+import org.sonatype.nexus.repository.maven.internal.storage.MavenStorageFacetImpl
+import org.sonatype.nexus.repository.search.SearchFacet
+import org.sonatype.nexus.repository.security.SecurityHandler
+import org.sonatype.nexus.repository.storage.StorageFacetImpl
+import org.sonatype.nexus.repository.types.HostedType
+import org.sonatype.nexus.repository.view.ConfigurableViewFacet
+import org.sonatype.nexus.repository.view.Route
+import org.sonatype.nexus.repository.view.Router
+import org.sonatype.nexus.repository.view.ViewFacet
+import org.sonatype.nexus.repository.view.handlers.TimingHandler
+import org.sonatype.nexus.repository.view.matchers.token.TokenMatcher
+
+import static org.sonatype.nexus.repository.http.HttpHandlers.notFound
+
+/**
+ * Maven 2 group repository recipe.
+ *
+ * @since 3.0
+ */
+@Named(Maven2GroupRecipe.NAME)
+@Singleton
+class Maven2GroupRecipe
+    extends RecipeSupport
+{
+  static final String NAME = "maven2-group"
+
+  @Inject
+  Provider<Maven2SecurityFacet> securityFacet
+
+  @Inject
+  Provider<ConfigurableViewFacet> viewFacet
+
+  @Inject
+  Provider<StorageFacetImpl> storageFacet
+
+  @Inject
+  Provider<SearchFacet> searchFacet
+
+  @Inject
+  TimingHandler timingHandler
+
+  @Inject
+  SecurityHandler securityHandler
+
+  @Inject
+  Provider<MavenStorageFacetImpl> mavenStorageFacet
+
+  @Inject
+  CoordinatesHandler coordinatesHandler
+
+  @Inject
+  MavenGroupHandler mavenGroupHandler
+
+  @Inject
+  Maven2GroupRecipe(@Named(HostedType.NAME) final Type type,
+                    @Named(Maven2Format.NAME) final Format format)
+  {
+    super(type, format)
+  }
+
+  @Override
+  void apply(@Nonnull final Repository repository) throws Exception {
+    repository.attach(securityFacet.get())
+    repository.attach(storageFacet.get())
+    repository.attach(searchFacet.get());
+    repository.attach(mavenStorageFacet.get())
+    repository.attach(configure(viewFacet.get()))
+  }
+
+  private ViewFacet configure(final ConfigurableViewFacet facet) {
+    Router.Builder builder = new Router.Builder()
+
+    builder.route(new Route.Builder()
+        .matcher(new TokenMatcher("/{name:.+}"))
```

Comment:
Unsure why we aren't using the token matcher to match the m2 coordinates, that is what the token system is meant to do.